### PR TITLE
Add WA Teams integration

### PR DESCRIPTION
**Pull request by Lennonka**
A new integration! At last...
It's been tested and code-reviewed by our developer as usual.

### CODE REVIEW

File: wa-integrations/teamsbot/routes/bot.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    await axios.post(strURL, inputBody, { headers: tyntecHeaders });
+    res.sendStatus(204);
+  } catch (e) {
+    handleError(e);
+    res.sendStatus(500);
+  }
+}
+
+// Teams -> WhatsApp
+router.post('/forwardwhatsapp', function (req, res, next) {
```
Comment (coders-kitchen) [inline wa-integrations/teamsbot/routes/bot.js:55]: Shouldn't it be `/forwardWhatsApp` ?

File: wa-integrations/teamsbot/routes/bot.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        await sendWAMessage(teamsUsers.get(turnContext.activity.from.id).assignedWhatsAppNumber, turnContext.activity.text, res);
+      }
+
+      next();
+    }
+  });
+});
+
+// WhatsApp -> Teams
+router.post('/forwardteams', async (req, res) => {
```
Comment (coders-kitchen) [inline wa-integrations/teamsbot/routes/bot.js:91]: And vice versa `/forwardTeams` :)

File: wa-integrations/teamsbot/routes/bot.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  console.log('Send input from WhatsApp to teams user.');
+  console.log('Received WA message: ', req.body.content.text);
+  console.log('From number: ', req.body.from);
+  var strWhatsAppNumber = req.body.from;
+
+  // Check if this WA number is already assigned to a conversation with a teams user
+  var blnFoundNumber = false;
```
Comment (coders-kitchen) [inline wa-integrations/teamsbot/routes/bot.js:98]: for the sake of readability I would propose to change the name to `conversationFound`

File: wa-integrations/teamsbot/routes/bot.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }
+  });
+});
+
+// WhatsApp -> Teams
+router.post('/forwardteams', async (req, res) => {
+  console.log('Send input from WhatsApp to teams user.');
+  console.log('Received WA message: ', req.body.content.text);
+  console.log('From number: ', req.body.from);
+  var strWhatsAppNumber = req.body.from;
```
Comment (coders-kitchen) [inline wa-integrations/teamsbot/routes/bot.js:95]: First time I see the the type on the variable present.

Couldn't this be simply `whatsAppNumber`?

**Review by coders-kitchen — COMMENTED** [review]
Please take a look at the variable namings.

Specifically two cases : 

 - abbreviations
 - type information 

IIRC this breaks the existing conventions

**Comment by Lennonka** [comment]
Hi! Sorry, I missed the review comments. You're right, I fixed the naming of vars and endpoints. Hopefully it's alright now.

**Pull request merged**